Sources/CircularProgress/protocol/MouseTrackable.swift

```diff
@@ -0,0 +1,25 @@
+#if os(macOS)
+import Cocoa
+/**
+Makes it possible to utilize mouseOver/mouseOut functionality on a view (Supports custom bezier path)
+*/
+protocol MouseTrackable:class{
+	var path:CGPath? {get}
+}
+extension MouseTrackable where Self:NSView{
+	/**
+	* Creates a new tracking area
+	*/
+	internal func createTrackingArea(_ options:NSTrackingArea.Options = [.activeInActiveApp, .mouseEnteredAndExited, .mouseMoved]){
```

Comment:
`internal` is the default. You don't need to specify that.

Reply:
I can make the extension internal, which would make all encapsulated methods also internal? I could be wrong but internal for me means, only accessible within the .framework.

Reply:
> All entities in your code (with a few specific exceptions, as described later in this chapter) **have a default access level of internal if you don’t specify an explicit access level yourself**. As a result, in many cases you don’t need to specify an explicit access level in your code. - https://docs.swift.org/swift-book/LanguageGuide/AccessControl.html

---

> I can make the extension internal, which would make all encapsulated methods also internal?

Nah. I prefer putting access levels on methods themselves, not the whole extension.

Reply:
✅